core/controllers/base.py

```diff
@@ -252,8 +263,13 @@ def dispatch(self):
             self.redirect('/logout?redirect_url=%s' % self.request.uri)
             return
 
-        if self.payload is not None and self.REQUIRE_PAYLOAD_CSRF_CHECK:
-            try:
+        try:
+            if 'csrf' in self.request.uri:
+                if self.gae_id and not self.user_id:
```

Comment:
Maybe explain the rationale behind this a bit.

Reply:
Since all logged user requests rely on CSRF token we want to block only this request for now.